resource_catalogue_fastapi/__init__.py

```diff
@@ -84,30 +88,58 @@ def opa_dependency(request: Request, path_params: dict = Depends(get_path_params
             raise HTTPException(status_code=403, detail="Access denied")
 
 
+class OrderStatus(Enum):
+    ORDERABLE = "orderable"
+    ORDERED = "ordered"
+    PENDING = "pending"
+    SHIPPING = "shipping"
+    SUCCEEDED = "succeeded"
+    FAILED = "failed"
+    CANCELED = "canceled"
+
+
 class ItemRequest(BaseModel):
     url: str
+    extra_data: Optional[Dict[str, Any]] = Field(default_factory=dict)
 
 
-def upload_nested_files(url: str, workspace: str) -> list:
+def upload_nested_files(
+    url: str, workspace: str, catalog_name: str = "saved-data", order_status: Optional[str] = None
+) -> Tuple[Dict[str, List[str]], Optional[str]]:
     """Upload a file along with any nested files to a workspace"""
     files_to_add = get_nested_files_from_url(url)
-    keys = []
+    keys = {"added_keys": [], "updated_keys": []}
+    ordered_item_key = None
     for url_to_add in files_to_add:
         logger.info(f"Adding url {url_to_add}")
         body = get_file_from_url(url_to_add)
 
-        workspace_key = f"{workspace}/saved-data/{url_to_add.split('/', 9)[-1]}"
+        workspace_key = f"{workspace}/{catalog_name}/{url_to_add.split('/', 9)[-1]}"
```

Comment:
For clarity, I'm wondering if it would be worth making this a variable of its own to have its own name - I'm not sure what the significance is of 9.

Or a comment, but comments often get out of date

Reply:
Added comment and moved split to named variable